src/nodes.coffee

```diff
@@ -1532,6 +1532,9 @@ exports.Param = class Param extends Base
         obj = obj.variable
       # * assignments within destructured parameters `{foo:bar}`
       if obj instanceof Assign
+        # ... possibly with a default value
+        if obj.value instanceof Assign
```

Comment:
It’s actually not a bug. It means `({a = (b = 1)}) ->`. That’s valid in JavaScript too.